packages/styled-system/src/utils/transforms.js

```diff
@@ -93,16 +108,28 @@ export const positiveOrNegative = (scale, value, options) => {
    * ```
    */
   if (typeof value === 'number' && Number.isFinite(value)) {
-    const isNegative = (value < 0);
-    if (!isNegative) {
+    const absoluteValue = Math.abs(value);
+    const isNonNegative = !(value < 0);
+
+    // Return the result if the value is non-negative or if the scale object does not contain the absolute value
+    if (isNonNegative || !Object.prototype.hasOwnProperty.call(scale, absoluteValue)) {
```

Comment:
I have been considering whether to use `Object.hasOwn` or `Object.prototype.hasOwnProperty.call`.

https://caniuse.com/mdn-javascript_builtins_object_hasown
![image](https://github.com/trendmicro-frontend/tonic-ui/assets/447801/c85b08b4-b8dd-4ff1-b3ba-04c1fecd4020)

`Object.hasOwn` was introduced around 2021. Since it's still quite new and the `styled-system` package acts as a common library rather than an app, I might choose to use the legacy `Object.prototype.hasOwnProperty.call`  for now to ensure that the library functions correctly in older environments.

Reply:
Got it, thanks for sharing that context, and i think it's worthwhile to explore alternative options to polyfil new JS features for older browser, just so you can get access to new and improved features